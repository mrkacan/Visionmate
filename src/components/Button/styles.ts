import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  wrapper: {
    width: '100%',
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 10,
    backgroundColor: '#088ab9',
    height: 48,
    marginBottom: 0,
  },
  text: {
    fontSize: 20,
    color: '#fff',
    fontWeight: 'bold',
  },
  fitContent: {
    width: 100,
    margin: 0,
  },
});

export default styles;
