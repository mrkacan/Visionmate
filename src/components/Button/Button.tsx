import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import styles from './styles';

type ButtonProps = {
  children: string;
  fitContent?: boolean;
  onPress: () => void;
}

const Button: React.FC<ButtonProps> = ({ onPress, children, fitContent = false }) => (
  <TouchableOpacity
    onPress={onPress}
    style={[styles.wrapper, fitContent && styles.fitContent]}
  >
    <Text style={styles.text}>
      {children}
    </Text>
  </TouchableOpacity>
);

export default Button;
