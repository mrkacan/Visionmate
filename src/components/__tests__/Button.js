import * as React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button/Button';

it('renders correctly', () => {
  const tree = renderer.create(<Button onPress={() => true}>Test button</Button>).toJSON();

  expect(tree).toMatchSnapshot();
});
