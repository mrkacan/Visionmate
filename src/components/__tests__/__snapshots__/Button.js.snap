// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#088ab9",
      "borderRadius": 10,
      "height": 48,
      "justifyContent": "center",
      "marginBottom": 0,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Test button
  </Text>
</View>
`;
